### PR TITLE
HDDS-2498. Fix sonar issues found in StorageContainerManager.

### PR DESCRIPTION
**Pull request by swagle**
## What changes were proposed in this pull request?
Fix all the issues except the TODO items.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-2498

## How was this patch tested?
Only syntactic changes, waiting on UT and acceptance runs.

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java [hunk 1]
```diff
@@ -678,25 +673,21 @@ public SCMSecurityProtocolServer getSecurityProtocolServer() {
...
-                      ContainerStat stat = removalNotification.getValue();
-                      // remove invalid container report
-                      metrics.decrContainerStat(stat);
-                      if (LOG.isDebugEnabled()) {
-                        LOG.debug("Remove expired container stat entry for " +
-                            "datanode: {}.", removalNotification.getKey());
-                      }
+            .removalListener((
+                RemovalListener<String, ContainerStat>) removalNotification -> {
+                  synchronized (containerReportCache) {
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java]: While Sonar did not flag it, IDEA warns about synchronization on non-final `containerReportCache`.  Can you please consider making it final?

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java [hunk 2]
```diff
@@ -678,25 +673,21 @@ public SCMSecurityProtocolServer getSecurityProtocolServer() {
    *
    * @param conf
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java]: Leftover `@param`.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java [hunk 3]
```diff
@@ -763,22 +754,27 @@ public String getDatanodeRpcPort() {
...
     commandWatcherLeaseManager.start();
     getClientProtocolServer().start();
 
-    LOG.info(buildRpcServerStartMessage("ScmBlockLocationProtocol RPC " +
-        "server", getBlockProtocolServer().getBlockRpcAddress()));
+    if (LOG.isInfoEnabled()) {
+      LOG.info(buildRpcServerStartMessage("ScmBlockLocationProtocol RPC " +
+          "server", getBlockProtocolServer().getBlockRpcAddress()));
+    }
     getBlockProtocolServer().start();
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java]: Are these messages needed at all?  Each server's `start()` method logs the same info.  Eg:

```
INFO server.StorageContainerManager: ScmBlockLocationProtocol RPC server is listening at /0.0.0.0:9863
INFO server.SCMBlockProtocolServer: RPC server for Block Protocol is listening at /0.0.0.0:9863
```

@anuengineer Is it for safety?

Comment (swagle) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java]: Unfortunately, slf4j does not yet support Supplier based parameters: https://jira.qos.ch/browse/SLF4J-371

Kept things unchanged and got rid of the sonar warning.

**Comment by swagle** [comment]
I've added a TODO because:
1. The synchronization will not protect double decrement, seems wrongly intended locking behavior.
2. IMO removal listener will also be triggered for cache invalidation since we are changing metric values need to gather more context to figure out what was the cause and effect here.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java [hunk 4]
```diff
@@ -675,31 +671,28 @@ public SCMSecurityProtocolServer getSecurityProtocolServer() {
...
+                  // TODO: Are we doing the right thing here?
+                  // remove invalid container report
+                  metrics.decrContainerStat(stat);
+                }
+                if (LOG.isDebugEnabled()) {
+                  LOG.debug("Remove expired container stat entry for " +
+                      "datanode: {}.", removalNotification.getKey());
+                }
+              }
+            })
```
Comment (dineshchitlangia) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java]: L682 - L694 : Checkstyle issue regarding incorrect indentation

**Pull request merged**